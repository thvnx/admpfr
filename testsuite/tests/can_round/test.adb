with Ada.Text_IO; use Ada.Text_IO;
with Admpfr;      use Admpfr;

procedure Test is
   X, Y : Mpfloat;
   Px   : constant Precision := 53;
   Py   : constant Precision := 50;
   OK   : Boolean;
   Err  : Exponent;
begin
   X.Set_Prec (Px);
   Y.Set_Prec (Py);

   for R1 in RNDN .. RNDD loop

      X.Const_Pi (R1);
      Put ("R1=" & R1'Image & " approx=");
      Out_Str (Standard_Output, X, 2);
      Put (ASCII.LF);

      for R2 in RNDN .. RNDD loop
         Err := Exponent (X.Get_Prec);
         OK := X.Can_Round (Err, R1, R2, Py);
         Put ("R2=" & R2'Image & " OK=" & OK'Image);
         if OK then
            Y.Set (X, R2);
            Put ("   ");
            Out_Str (Standard_Output, Y, 2);
         end if;
         Put (ASCII.LF);
      end loop;
   end loop;
end Test;
