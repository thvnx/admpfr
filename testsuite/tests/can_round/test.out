R1=RNDN approx=1.1001001000011111101101010100010001000010110100011000e1
R2=RNDN OK=TRUE   1.1001001000011111101101010100010001000010110100011e1
R2=RNDZ OK=FALSE
R2=RNDU OK=FALSE
R2=RNDD OK=FALSE
R1=RNDZ approx=1.1001001000011111101101010100010001000010110100011000e1
R2=RNDN OK=TRUE   1.1001001000011111101101010100010001000010110100011e1
R2=RNDZ OK=TRUE   1.1001001000011111101101010100010001000010110100011e1
R2=RNDU OK=FALSE
R2=RNDD OK=TRUE   1.1001001000011111101101010100010001000010110100011e1
R1=RNDU approx=1.1001001000011111101101010100010001000010110100011001e1
R2=RNDN OK=TRUE   1.1001001000011111101101010100010001000010110100011e1
R2=RNDZ OK=TRUE   1.1001001000011111101101010100010001000010110100011e1
R2=RNDU OK=FALSE
R2=RNDD OK=TRUE   1.1001001000011111101101010100010001000010110100011e1
R1=RNDD approx=1.1001001000011111101101010100010001000010110100011000e1
R2=RNDN OK=TRUE   1.1001001000011111101101010100010001000010110100011e1
R2=RNDZ OK=TRUE   1.1001001000011111101101010100010001000010110100011e1
R2=RNDU OK=FALSE
R2=RNDD OK=TRUE   1.1001001000011111101101010100010001000010110100011e1

